### DISCUSSION TITLE
imports

### DISCUSSION THREAD

**Discussion by borische1322** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#56] [Q&A]
i just noticed that in my gameboardcontoller.java the imports has been modified on its on when i tried to create classes like positions or gems but once i change it back to the original import i can no longer create position or gem classes, are we not supposed to create position or gem classes in gameboardcontroller.java?

**Comment by borische1322** [comment]
like for example the original import is 
import hk.ust.cse.comp3021.pa1.model.Direction;
import hk.ust.cse.comp3021.pa1.model.GameBoard;
import hk.ust.cse.comp3021.pa1.model.MoveResult;
import org.jetbrains.annotations.NotNull;

but now it has changed to 
import hk.ust.cse.comp3021.pa1.model.*;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Objects;

and i didnt even motify it

**Reply by Derppening** [reply]
IntelliJ automatically converts multiple imports from the same package into a wildcard import (which is the `hk.ust.cse.comp3021.pa1.model.*` import you see).

I don't understand how this is a problem.

**Reply by borische1322** [reply]
oh so it wont deduct some points?

**Reply by Derppening** [reply]
Well as long as CheckStyle does not complain about it, then there is no reason for me to deduct points.
